lib/spack/spack/binary_distribution.py

```diff
@@ -346,21 +347,23 @@ def update(self, with_cooldown=False):
                     with_cooldown
                     and ttl > 0
                     and cached_mirror_url in self._last_fetch_times
-                    and now - self._last_fetch_times[cached_mirror_url] < ttl
+                    and now - self._last_fetch_times[cached_mirror_url][0] < ttl
                 ):
-                    # The fetch worked last time, so don't error
-                    all_methods_failed = False
+                    if self._last_fetch_times[cached_mirror_url][1]:
+                        # The fetch worked last time, so don't error
+                        all_methods_failed = False
```

Comment:
Can you add a comment here like:

```
if self._last_fetch_times[cached_mirror_url][1]:
    # The fetch worked last time, so don't error
    all_methods_failed = False
# else: we've already tried once within the cooldown period;
# re-report the fact that we failed
```

Reply:
How's this:
```python
# We're in the cooldown period, don't try to fetch again
# If the fetch succeeded last time, consider this update a success, otherwise
# re-report the error here
if self._last_fetch_times[cached_mirror_url][1]:
  all_methods_failed = False
```

Reply:
This works fine, thanks!